includes/jobs/GlobalNewFilesMoveJob.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+use MediaWiki\MediaWikiServices;
+
+class GlobalNewFilesMoveJob extends Job implements GenericParameterJob {
+	private $title;
+
+	private $newTitle;
+
+
+	/**
+	 * @param array $params
+	 */
+	public function __construct( array $params ) {
+		parent::__construct( 'GlobalNewFilesMoveJob', $params );
+
+		$this->title = $params['title'];
+		$this->newTitle = $params['newtitle'];
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function run() {
+		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'globalnewfiles' );
+
+		if ( !$this->title->inNamespace( NS_FILE ) ) {
+			return true;
+		}
```

Comment:
i would suggest keeping those checks in the event handlers - given the title object is already loaded that's a cheap check and it avoids queuing unnecessary jobs on most operations, when those are only needed when done on the file ns

Reply:
Done. Thanks!